src/Microsoft.Data.SqlClient/netfx/src/Microsoft/Data/SqlClient/TdsParser.cs

```diff
@@ -3341,7 +3341,7 @@ private bool TryProcessDone(SqlCommand cmd, SqlDataReader reader, ref RunBehavio
             // _pendingData set by e.g. 'TdsExecuteSQLBatch'
             // _hasOpenResult always set to true by 'WriteMarsHeader'
             //
-            if (!stateObj._attentionSent && !stateObj._pendingData && stateObj._hasOpenResult)
+            if (!stateObj._attentionReceived && !stateObj._pendingData && stateObj._hasOpenResult)
```

Comment:
Why did not use the same naming as NetCore?

Reply:
Legacy code.. I think in NetCore side improvements were made that brought these differences.
It's out of scope of this PR, so I won't bother making too many changes here. Maybe something for future!